common/cache/src/main/java/org/thingsboard/server/cache/RedisTbTransactionalCache.java

```diff
@@ -130,6 +151,9 @@ public void evict(Collection<K> keys) {
 
     @Override
     public void evictOrPut(K key, V value) {
+        if (!cacheEnabled) {
+            return;
+        }
```

Comment:
We also need to make sure `getAndPutInTransaction` doesn't use cache. You'll have to modify `newTransactionForKey` and probably `RedisTbCacheTransaction`.